pallets/funding/src/functions.rs

```diff
@@ -1309,8 +1332,22 @@ impl<T: Config> Pallet<T> {
 			InvestorType::Professional => project_metadata.contributing_ticket_sizes.professional,
 			InvestorType::Retail => project_metadata.contributing_ticket_sizes.retail,
 		};
+		let max_multiplier = match investor_type {
+			InvestorType::Retail => {
+				RetailParticipations::<T>::mutate(&did, |project_participations| {
+					if project_participations.contains(&project_id).not() {
+						// We don't care if it fails, since it means the user already has access to the max multiplier
+						let _ = project_participations.try_push(project_id);
+					}
+					retail_max_multiplier_for_participations(project_participations.len() as u8)
+				})
```

Comment:
If this "participation" only counts if a Retail investor participates during contribution round and not evaluation, we could do a check on the `ContributionBoughtUSD` if balance == 0 for (project_id, did) to determine if its the first participation for this project_id. This way we do not have to store a full vec of project_ids that retail investor participated in, but just a u32 or something.